pex/pex_builder.py

```diff
@@ -324,11 +324,18 @@ def add_dist_location(self, dist, name=None):
     supported by setuptools/pkg_resources.
     """
     self._ensure_unfrozen('Adding a distribution')
-    bdist = DistributionHelper.distribution_from_path(dist)
-    if bdist is None:
-      raise self.InvalidDistribution('Could not find distribution at %s' % dist)
-    self.add_distribution(bdist, dist_name=name)
-    self.add_requirement(bdist.as_requirement())
+    dist_path = dist
+    if os.path.isfile(dist_path) and dist_path.endswith('.whl'):
+      dist_path = os.path.join(safe_mkdtemp(), os.path.basename(dist))
+      spawn_install_wheel(
+        wheel=dist,
+        install_dir=dist_path,
+        target=DistributionTarget.for_interpreter(self.interpreter)
+      ).wait()
```

Comment:
This part I'm unclear on. Is this an optimization or necessary to get things working?

Reply:
Its needed to kick our custom zipped-wheel finder to the curb. It's not an optimization since all wheels added to a pex are installed, not zipped (this goes back to a few commits after wheel support was added way back). Really, the zipped wheel finder support was only ~needed in tests and now that's killed too.